src/software/ai/evaluation/time_to_travel.cpp

```diff
@@ -0,0 +1,70 @@
+#include <cmath>
+
+#include "software/ai/evaluation/time_to_travel.h"
+
+Duration getTimeToTravelDistance(const double distance, const double max_velocity,
+                                 const double max_acceleration,
+                                 const double initial_velocity,
+                                 const double final_velocity)
+{
+    // Bound all values to be realistic
+    double d_total = std::max(0.0, distance);
+    double v_max   = std::max(0.0, max_velocity);
+    double v_i     = std::clamp(initial_velocity, -max_velocity, max_velocity);
+    double v_f     = std::clamp(final_velocity, 0.0, max_velocity);
```

Comment:
Initial velocity can be negative since initially you may be driving away from your destination, however, final velocity didn't make much sense to me when its negative. I think your velocity would be either 0 or positive to pass through the destination, if it's negative (the robot is coming back), I would imagine that you would calculate the time to travel the _path_ in two steps:
1) going to the first destination and stopping at it (Vf = 0) 
2) Starting with Vi = 0 at first destination and going to the second destination point which would bring the robot back.

That was the assumption I went off of.

Reply:
that makes sense to me now. The velocity is relative to the robot; in my mind I was imagining the that a robot travelling in the -X direction would have a negative velocity (aka velocity relative to the field)

Reply:
Added the constraint to the header